DfE.FindInformationAcademiesTrusts.Data.AcademiesDb/Repositories/DataSourceRepository.cs

```diff
@@ -36,6 +39,38 @@ public async Task<DataSource> GetMisEstablishmentsUpdatedAsync()
         return new DataSource(Source.Mis, lastEntry.Modified.Value, UpdateFrequency.Monthly);
     }
 
+    /// <summary>
+    /// Retrieve MSTR data source information by switching on the Source value.
+    /// </summary>
+    private async Task<DataSource> GetDataSourceFromMstr(Source source)
+    {
+        // Default to Daily, but adjust if needed
+        var updateFrequency = UpdateFrequency.Daily;
+        DateTime? lastDataRefresh = null;
+
+        lastDataRefresh = source switch
+        {
+            Source.Prepare => await academiesDbContext.MstrAcademyTransfers
+                                .Where(t => t.InPrepare == "Yes")
```

Comment:
As we are querying by this nullable db field, it would be good to add "is not null" to our default EF query filters. This gives a minor performance improvement by allowing SQL to treat the column as a non-null column whilst querying it.

Reply:
(note, comment applies to all nullable columns we are filtering by on the new tables)